### ISSUE TITLE
[🚀 Feature]: Mini-manager for Selenium Manager in bindings

### ISSUE THREAD

**Issue by bonigarcia** [SeleniumHQ/selenium#11693]
### Feature and motivation

Currently, the Selenium Manager binaries are shipped in each binding release. There is another alternative to this, which may be the most costly at the beginning but more optimal in the long run. It consists on implementing a mini-manager logic in each binding to download the latest Selenium Manager release to the local machine (to the `.cache/selenium` folder). To do this, we need the following:

- Make a release of Selenium Manager in GitHub (together with the rest of each release artifacts).
- Each binding version (e.g., 4.9.0) only uses a fixed version of Selenium Manager (let's say the same number, 4.9.0).
- Download the proper Selenium Manager binary, i.e., for the local operating system (at this moment, and maybe for different platforms in the future), to the local cache.
- Invoke the local Selenium Manager (as it is done already) in each binding.

The pros of this approach are:
- We avoid packaging the bindings in each Selenium release.
- We can easily add more Selenium Manager platforms (e.g., mac-arm64).
- We have a figure about the usage of Selenium Manager out of the box (looking at the GitHub releases)

The cons of this approach are:
- We need to implement (and maintain) the mini-manager in each binding.

### Usage example

n/a

**Comment by titusfortner** [comment]
Yeah, I think the con outweighs the pros here. Not needing to manage an http client with request management in each of the bindings was a big part of why we wanted to implement this functionality in Rust in the first place.

**Comment by diemol** [comment]
I think Titus is right. What I wonder is if we can implement what we discussed once, @bonigarcia, which is to have Selenium Manager update itself? Or maybe have a very small Rust binary that does the job of being a mini manager?

**Comment by titusfortner** [comment]
Yeah, that's what I was wondering; if we could have a super small Selenium Manager Manager whose only responsibility was to figure out the right Selenium Manager to load if one wasn't already found. We could have the 5 binaries for the various architectures or whatever, they'd just all be like 1 MB instead of 5? Kind of depends on how small we could make it...

Even this feels like a low priority, though. I don't think package size is a major limitation.

**Comment by bonigarcia** [comment]
> I think Titus is right. What I wonder is if we can implement what we discussed once, @bonigarcia, which is to have Selenium Manager update itself?

It can be done, but I discarded that idea when we decided to ship Selenium Manager within the bindings distributions (since the update happens in each release).

> Or maybe have a very small Rust binary that does the job of being a mini manager?

Having a Selenium Meta-Manager is an interesting idea, although I am a bit skeptical about that. Since even using the minimum number of dependencies possible on the Rust side, the Meta-Manager would require `tokio`, `reqwest` and `tempfile` for download, and I suppose we will need to maintain some metadata about the downloaded Selenium Manager, and for that, we would need at least `serde` and `serde_json`. Overall, the weight of the Meta-Manager will be a bit less than Selenium Manager, but I suspect not that much. In summary, maybe the cake is not worth the candle.
